src/PointCloud/Core/PointCloudOctant.cs

```diff
@@ -193,6 +194,56 @@ public bool Intersects(double3 point)
             (point.z >= Min.z && point.z <= Max.z);
         }
 
+
+        /// <summary>
+        /// Returns true if the shape is completely inside or is intersecting this <see cref="PointCloudOctant"/>
+        /// </summary>
+        /// <param name="invSphereModel">The inverse model matrix which is currently used to render the <see cref="Sphere"/> (can e. g. be displayed as an ellipsoid)</param>
+        /// <returns></returns>
+        public bool InsideOrIntersectingSphere(double4x4 invSphereModel)
```

Comment:
When I think about it now the method should probably be named `InsideOrIntersectingEllipsoid`. For `InsideOrIntersectingSphere` we only need the center and radius of the sphere and don't need the matrix multiplication that comes with the inverse transformation.